wp_org_api/src/plugin_directory.rs

```diff
@@ -0,0 +1,542 @@
+    })
+}
+
+#[cfg(test)]
+mod tests {
+    use super::*;
+    use rstest::*;
+    use std::path::PathBuf;
+
+    #[fixture]
+    fn plugins_dir() -> PathBuf {
+        std::fs::canonicalize("../target/wordpress-org-plugin-directory").unwrap()
+    }
+
+    #[fixture]
+    fn plugin_info_files(plugins_dir: PathBuf) -> Vec<PathBuf> {
+        println!(
+            "Reading plugin information files from {:?}...",
+            &plugins_dir
+        );
+
+        let mut files = vec![];
+        for entry in std::fs::read_dir(plugins_dir).unwrap() {
+            let entry = entry.unwrap();
+            if entry.file_type().unwrap().is_file()
+                && entry.path().extension().and_then(|f| f.to_str()) == Some("json")
+            {
+                files.push(entry.path());
+            }
+        }
+        files
+    }
+
+    fn parse_plugin(slug: &str) -> Result<PluginInformation, serde_json::Error> {
+        let file = plugins_dir().join(format!("{}.json", slug));
+        let content = std::fs::read_to_string(file).unwrap();
+        serde_json::from_str::<PluginInformation>(&content)
+    }
+
+    #[rstest]
+    fn parse_plugin_info(plugin_info_files: Vec<PathBuf>) {
+        let results: HashMap<&PathBuf, _> =
+            plugin_info_files
+                .iter()
+                .fold(HashMap::new(), |mut results, file| {
+                    let info = std::fs::read_to_string(file).unwrap();
+                    let result = serde_json::from_str::<PluginInformation>(&info);
+                    results.insert(file, result);
+                    results
+                });
+
+        let success: Vec<_> = results
+            .iter()
+            .filter(|(_, result)| result.is_ok())
+            .collect();
+
+        let failed: Vec<_> = results
+            .iter()
+            .filter(|(_, result)| result.is_err())
+            .map(|(file, _)| file)
+            .collect();
+        if !failed.is_empty() {
+            println!("Failed to parse the following files:");
+            for file in &failed {
+                println!("- {:?}", file.file_name().unwrap());
+            }
+        }
+
+        assert_eq!(
+            success.len(),
+            results.len(),
+            "{} out of {} files parsed successfully",
+            success.len(),
+            results.len()
+        );
+        assert!(
+            failed.is_empty(),
+            "Failed to parse {} out of {} files",
+            failed.len(),
+            results.len()
+        );
+    }
+
+    #[rstest]
+    #[case("jetpack", "https://profiles.wordpress.org/automattic/")]
+    #[case("appmysite", "https://profiles.wordpress.org/appmysite/")]
+    #[case("superlinks", "")]
+    fn test_property_author_profile(#[case] slug: &str, #[case] value: &str) {
+        let result = parse_plugin(slug);
+        assert!(
+            result.is_ok(),
+            "Failed to parse plugin {:?}: {:?}",
+            slug,
+            result.err()
+        );
```

Comment:
This seems to be replicated for all tests. How do you feel about `parse_plugin` handling the assertion and returning `PluginInformation`?